liquigraph-core/src/main/java/org/liquigraph/core/io/lock/ShutdownTask.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.io.lock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.UUID;
+
+/**
+ * Lock-related cleanup task for {@link LockableConnection}
+ */
+final class ShutdownTask implements Runnable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShutdownTask.class);
+
+    private final Connection connection;
+    private final UUID uuid;
+
+    public ShutdownTask(Connection connection, UUID uuid) {
+        this.connection = connection;
+        this.uuid = uuid;
+    }
+
+    @Override
+    public void run() {
```

Comment:
Why not just call a package-private `releaseLock()` method on `LockableConnection`, so that all the lock management is done in one place and the Cypher queries are next to each other? Since you need the `Connection`, you might as well reference it as a `LockableConnection` and lose the `uuid` field.